tests/shared/src/test/scala/org/http4s/StaticFileSuite.scala

```diff
@@ -297,7 +298,8 @@ class StaticFileSuite extends Http4sSuite {
   if (Platform.isJvm)
     test("Read from a URL") {
       val url = getClass.getResource("/lorem-ipsum.txt")
-      val expected = scala.io.Source.fromURL(url, "utf-8").mkString
+      val expected =
+        AutoCloseableResource.resource(scala.io.Source.fromURL(url, "utf-8"))(_.mkString)
```

Comment:
This is the only `StaticFileSuite` test using `AutoCloseableResource`, and it is JVM-only :)

Reply:
Oh, shame on me. `Platform.isJvm` 😄 I didn't expect to see this. Maybe it's better to split it into two test suites?

Reply:
If this file had been split, then this merge you just did would have been tripped up here!

Reply:
Although I agree, when the days of merges are over we should indeed split files where appropriate :)

Reply:
What I actually don't get is why the `testingJS` module compiles. Or it'll fall in the runtime (if some code will work with `AutoCloseable`)?

Reply:
Because `AutoCloseable` is supported on Scala.js :)
https://github.com/scala-js/scala-js/blob/main/javalanglib/src/main/scala/java/lang/AutoCloseable.scala

Reply:
But in general, yes, Scala.js code can compile, but that doesn't mean it will link/run. For example `scala.io.Source.fromURL` is not supported on Scala.js, but because it's in the if-JVM statement, the Scala.js linker is able to remove that code before running it.

Reply:
Good to know, thanks. It seems we need to move the `AutoCloseableResource` in the JVM module as well as separate that JVM-specific test-suite (otherwise, it will not compile). What do you think?

Reply:
I don't understand. The CI is green, and I don't see any reason it shouldn't be?

Reply:
Sorry, I'm confused a bit too. If we can leave it as is, I'll be pretty much glad :)

Reply:
Everything is perfect :) sorry I couldn't explain it very well.